### ISSUE TITLE
Mage Armors disappearing after duel

### ISSUE THREAD

**Issue by Liesara** [ccshiro/cc-buglist#2027]
Ironically enough this doesn't apply to the actual Mage Armor. It does apply to Frost and Ice Armor and to Molten Armor.

We were having some casual duels when we noticed Ice Armor dropping off way too early. So we tried every spell we could have possibly used and none killed the Armor. That's when we noticed it disappeared when the duel ended. After further research we found out that all Armor spells except the Mage Armor shared that behavior and also that it only happened on a melee kill (it does not happen on a hammer of wrath kill but we didn't test ranged physical kills). We theorized that the mechanic behind it was the armor procc being delayed until after the duel and not only the procc but the armor itself as well being killed therefore, since Mage Armor as the only one does not have a procc on being hit. So we also tested with Thorns and Retribution Aura and neither of them shared this behavior.

**Comment by Jumpka** [comment]
Tested it today and I think our theory about the proc being the problem is correct.
When we tested it with Liesara yesterday, Molten armor would disappear when I would win duel with a melee attack, but not with Hammer of Wrath. Today I did the same with the Molten Shields, which causes Molten Armor to reflect 75 fire damage even when being attacked by spells, and with this talent Molten Armor now disappears even after Hammer of Wrath duel win, so it has to be the proc. 

EDIT: Same thing happens with Lightning Shield.
